sdk/storage/azure-storage-blobs/CMakeLists.txt

```diff
@@ -2,72 +2,57 @@
 # SPDX-License-Identifier: MIT
 
 cmake_minimum_required (VERSION 3.13)
-
-set (AZURE_STORAGE_BLOB_HEADER
-    inc/azure/storage/blobs.hpp
-    inc/azure/storage/blobs/append_blob_client.hpp
-    inc/azure/storage/blobs/blob_batch_client.hpp
-    inc/azure/storage/blobs/blob_client.hpp
-    inc/azure/storage/blobs/blob_container_client.hpp
-    inc/azure/storage/blobs/blob_options.hpp
-    inc/azure/storage/blobs/blob_responses.hpp
-    inc/azure/storage/blobs/blob_sas_builder.hpp
-    inc/azure/storage/blobs/blob_service_client.hpp
-    inc/azure/storage/blobs/block_blob_client.hpp
-    inc/azure/storage/blobs/page_blob_client.hpp
-    inc/azure/storage/blobs/protocol/blob_rest_client.hpp
-    inc/azure/storage/blobs/version.hpp
-)
-
-set (AZURE_STORAGE_BLOB_SOURCE
-    src/append_blob_client.cpp
-    src/blob_batch_client.cpp
-    src/blob_client.cpp
-    src/blob_container_client.cpp
-    src/blob_sas_builder.cpp
-    src/blob_service_client.cpp
-    src/block_blob_client.cpp
-    src/page_blob_client.cpp
-)
-
-# CodeCoverage will validate requirements
-include(CodeCoverage)
-
-add_library(azure-storage-blobs ${AZURE_STORAGE_BLOB_HEADER} ${AZURE_STORAGE_BLOB_SOURCE})
-target_include_directories(azure-storage-blobs PUBLIC inc)
-target_link_libraries(azure-storage-blobs azure::storage::common)
+set(TARGET_NAME "azure-storage-blobs")
+project(${TARGET_NAME} LANGUAGES CXX)
+
+set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD_REQUIRED True)
+set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/../../../cmake-modules")
+
+include(az_vcpkg)
+include(az_version)
+include(DefinePlatform)
+include(DefineTransportAdapter)
+include(doxygen_common)
+include(global_compile_options)
+
+az_vcpkg_integrate()
+
+if(NOT AZ_ALL_LIBRARIES)
+  find_package(azure-storage-common-cpp CONFIG)
+  if(NOT azure-storage-common-cpp_FOUND)
+    find_package(azure-storage-common-cpp REQUIRED)
+  endif()
+endif()
+
+add_library(
+  ${TARGET_NAME}
+  src/append_blob_client.cpp
+  src/blob_batch_client.cpp
+  src/blob_client.cpp
+  src/blob_container_client.cpp
+  src/blob_sas_builder.cpp
+  src/blob_service_client.cpp
+  src/block_blob_client.cpp
+  src/page_blob_client.cpp
+  )
+
+# make sure that users can consume the project as a library.
+add_library(Azure::${TARGET_NAME} ALIAS ${TARGET_NAME})
+
+target_include_directories(
+  ${TARGET_NAME}
+    PUBLIC
+      $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/inc>
+      $<INSTALL_INTERFACE:include>
+      ${azure-storage-common-cpp_INCLUDE_DIRS}
+  )
+
+target_link_libraries(${TARGET_NAME} PUBLIC Azure::azure-storage-common)
```

Comment:
Is azure-storage-common expected to be used by end-users?
If not, this should be INTERFACE or PRIVATE

Reply:
> Is azure-storage-common expected to be used by end-users?

I think so. @Jinming-Hu, @katmsft - can you confirm what the intent is for azure-storage-common around end-user usage?

There are certain things like StorageException/etc. that are in azure-storage-common, so I suspect folks will reference it (probably not directly though). This package **doesn't only** contain common utility types/methods used by the other three storage packages.